sdk/src/main/java/com/bugsnag/android/SessionTracker.java

```diff
@@ -390,13 +393,17 @@ void updateForegroundTracker(String activityName, boolean activityStarting, long
             }
         }
         setChanged();
+        notifyNdkInForeground();
```

Comment:
I might have the wrong end of the elephant here, but since this is only triggered by activity lifecycle callbacks, does this change improve the value of `isInForeground` for the NDK?

Reply:
I've updated the PR description to explain why this approach is better following an in-person conversation, which I believe resolved this comment.